### PR TITLE
chore(watcher.js): A Watcher to that triggers for changes in file and rebuild the package

### PR DESCRIPTION
**Pull request by JayaKrishnaNamburu**
@alexnm this can help us in watching all packages at once and run ``tsc`` to build that package. I am using on my side to check if this creates any lag or something, since ``lerna run, build:watch --parallel`` is causing a lag for me.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -24,7 +24,8 @@
     "version": "lerna version",
     "publish": "lerna publish from-package",
     "local:publish": "lerna exec -- npm publish --registry http://localhost:4873",
-    "local:unpublish": "lerna exec -- npm unpublish -f --registry http://localhost:4873"
+    "local:unpublish": "lerna exec -- npm unpublish -f --registry http://localhost:4873",
+    "watch": "node watcher.js"
```
Comment (alexnm) [inline package.json]: can we replace the `dev` task with this if it works better?

File: watcher.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+const chokidar  = require('chokidar')
```
Comment (alexnm) [inline watcher.js]: can you please move this file into a `scripts` folder where we can further keep all setups and stuff like that?

Comment (JayaKrishnaNamburu) [reply watcher.js]: Sure 👍

File: package.json (hunk 1 above)
Comment (JayaKrishnaNamburu) [reply package.json]: Yeah yeah, currently I want to test for a couple of days. So kept as a side script

File: scripts/watcher.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  log(chalk.redBright.bold(`${filePath} is changed`))
+
+  const path = filePath.split('/')
+  const location = `${path[0]}/${path[1]}/`
+
+  exec(`tsc`, { cwd: location}, (err, stdout) => {
+    if (!err ||  err === null) {
+      log(chalk.greenBright(`Package ${path[1]} is re-built`))
+    } else {
+      log(chalk.white(err))
```
Comment (JayaKrishnaNamburu) [inline scripts/watcher.js]: @alexnm I added these lines to throw errors when compilation fails, I tested this our for some time and its smooth enough.

Comment (alexnm) [reply scripts/watcher.js]: cool, eager to test this out also :)

Comment (JayaKrishnaNamburu) [reply scripts/watcher.js]: I am just pushing one more commit, by adding ``incremental`` in **tsconfig.json** that will decrease the compilation time. Please try with that

**Comment by JayaKrishnaNamburu** [comment]
@alexnm you can check now

**Pull request merged**